flink-runtime/src/main/java/org/apache/flink/runtime/io/network/api/serialization/EventSerializer.java

```diff
@@ -107,10 +107,9 @@ else if (eventClass == CancelCheckpointMarker.class) {
 	 *
 	 * @param buffer the buffer to peak into
 	 * @param eventClass the expected class of the event type
-	 * @param classLoader the class loader to use for custom event classes
 	 * @return whether the event class of the <tt>buffer</tt> matches the given <tt>eventClass</tt>
 	 */
-	private static boolean isEvent(ByteBuffer buffer, Class<?> eventClass, ClassLoader classLoader) throws IOException {
+	private static boolean isEvent(ByteBuffer buffer, Class<?> eventClass) throws IOException {
```

Comment:
this change qualifies for a separate JIRA ticket, not just a hotfix

Reply:
You should also add a comment, that checking for custom events is not supported.